addons/strings/fnc_substring.sqf

```diff
@@ -28,18 +29,12 @@ SCRIPT(substring);
 
 // ----------------------------------------------------------------------------
 
-PARAMS_3(_string,_startIndex, _endIndex);
-private ["_arr", "_ret", "_i", "_x"];
-_arr = toArray _string;
-_ret = [];
+params["_string","_startIndex","_endIndex"];
+// Check if _start is Larger than _endIndex to Prevent Issues
+if (_startIndex > _endIndex) exitWith {""};
+// Calculate Differenz between _start and _end for select lenth value
+_endIndex = _endIndex + 1 - _startIndex;
+// Cut out String
+_ret = _string select [_startIndex, _endIndex];

```

Comment:
You can cut out the middleman here and just return `_string select [_startIndex, _endIndex]`